src/daft-functions-utf8/src/count_matches.rs

```diff
@@ -96,6 +99,75 @@ impl ScalarUDF for CountMatches {
     }
 }
 
+#[typetag::serde]
+impl ScalarUDF for CountMatchesRegex {
+    fn name(&self) -> &'static str {
+        "count_matches_regex"
+    }
+    fn call(&self, inputs: daft_dsl::functions::FunctionArgs<Series>) -> DaftResult<Series> {
+        let input = inputs.required((0, "input"))?;
+        let patterns = inputs.required((1, "patterns"))?;
+
+        let whole_words = inputs.optional("whole_words")?.map(|s| {
+            ensure!(s.data_type().is_boolean() && s.len() == 1, ValueError: "expected boolean literal for 'whole_words'");
+            Ok(s.bool().unwrap().get(0).unwrap())
+        }).transpose()?.unwrap_or(WHOLE_WORDS_DEFAULT_VALUE);
+        let case_sensitive = inputs.optional("case_sensitive")?.map(|s| {
+            ensure!(s.data_type().is_boolean() && s.len() == 1, ValueError: "expected boolean literal for 'case_sensitive'");
+            Ok(s.bool().unwrap().get(0).unwrap())
+        }).transpose()?.unwrap_or(CASE_SENSITIVE_DEFAULT_VALUE);
+
+        ensure!(patterns.len() == 1, ValueError: "Cannot set `patterns` in `count_matches_regex` to an Expression. Only string is currently supported.");
+
```

Comment:
Not blocking, but you should take a look at using the `#[derive(FunctionArgs)]` proc macro to do this validation instead of manually writing it for both `call` and `get_return_field`: https://github.com/Eventual-Inc/Daft/blob/main/src/common/macros/src/lib.rs#L7